rtt/base/TaskCore.hpp

```diff
@@ -337,6 +337,17 @@ namespace RTT
             return ee;
         }
 
+        /**
+         * The name of this TaskCore
+         *
+         * Used for debugging and/or tracing purposes
+         */
+        std::string getName() const;
+
+        /** Sets the name of this TaskCore
+         */
+        void setName(const std::string& name);
```

Comment:
Theoretically names can become inconsistent as a TaskContext's name can be changed via `tc->setName(...)` inherited from TaskCore and also via `tc->provides()->setName(...)`, which is the return value of `tc->getName()`. Please add an override of `setName()` in class TaskContext that sets both.

Or remove the `setName()` method completely in favor of the constructor overload. Changing the name of a component after its construction does not play nicely together with full port names anyway.

Reply:
I found that the CORBA `TaskContextProxy` changes the name of the underlying `TaskContext` after initial construction in [TaskContextProxy.cpp:151](https://github.com/orocos-toolchain/rtt/blob/48279d7539f26bf344f905ea411e6e6067db1a70/rtt/transports/corba/TaskContextProxy.cpp#L151) or [TaskContextProxy.cpp:182](https://github.com/orocos-toolchain/rtt/blob/48279d7539f26bf344f905ea411e6e6067db1a70/rtt/transports/corba/TaskContextProxy.cpp#L182).

Not a big issue because proxies are not supposed to be traced, but they do create (and immediately destroy) a thread and trigger the `thread_init` tracepoint.